modules/server/src/test/scala/navigate/server/tcs/TcsBaseControllerEpicsSpec.scala

```diff
@@ -569,11 +569,184 @@ class TcsBaseControllerEpicsSpec extends CatsEffectSuite {
+      assert(r1.m1GuideConfig.filename.connected)
+      assert(r1.m2Guide.connected)
+      assert(r1.m2GuideConfig.source.connected)
+      assert(r1.m2GuideConfig.beam.connected)
+      assert(r1.m2GuideConfig.filter.connected)
+      assert(r1.m2GuideConfig.samplefreq.connected)
+      assert(r1.m2GuideConfig.reset.connected)
+      assert(r1.m2GuideMode.connected)
+      assert(r1.m2GuideReset.connected)
+      assert(r1.mountGuide.mode.connected)
+      assert(r1.mountGuide.source.connected)
+      assert(r1.guiderGains.p1TipGain.connected)
+      assert(r1.guiderGains.p1TiltGain.connected)
+      assert(r1.guiderGains.p1FocusGain.connected)
+      assert(r1.guiderGains.p2TipGain.connected)
+      assert(r1.guiderGains.p2TiltGain.connected)
+      assert(r1.guiderGains.p2FocusGain.connected)
+      assert(r1.guiderGains.oiTipGain.connected)
+      assert(r1.guiderGains.oiTiltGain.connected)
+      assert(r1.guiderGains.oiFocusGain.connected)
+
+      assertEquals(r1.m1Guide.value.flatMap(Enumerated[BinaryOnOff].fromTag), BinaryOnOff.On.some)
+      assertEquals(r1.m1GuideConfig.source.value.flatMap(Enumerated[M1Source].fromTag),
+                   M1Source.Oiwfs.some
+      )
+      assertEquals(r1.m1GuideConfig.frames.value.flatMap(_.toIntOption), 1.some)
+      assertEquals(r1.m1GuideConfig.weighting.value, "none".some)
+      assertEquals(r1.m1GuideConfig.filename.value, "".some)
+      assertEquals(r1.m2Guide.value.flatMap(Enumerated[BinaryOnOff].fromTag), BinaryOnOff.On.some)
+      assertEquals(r1.m2GuideConfig.source.value.flatMap(Enumerated[TipTiltSource].fromTag),
+                   TipTiltSource.Oiwfs.some
+      )
+      assertEquals(r1.m2GuideConfig.beam.value, "B".some)
+      assertEquals(r1.m2GuideConfig.filter.value, "raw".some)
+      assertEquals(r1.m2GuideConfig.samplefreq.value.flatMap(_.toDoubleOption), 200.0.some)
+      assertEquals(r1.m2GuideConfig.reset.value.flatMap(Enumerated[BinaryOnOff].fromTag),
+                   BinaryOnOff.Off.some
+      )
+      assertEquals(r1.m2GuideMode.value.flatMap(Enumerated[BinaryOnOff].fromTag),
+                   BinaryOnOff.On.some
+      )
+      assertEquals(r1.m2GuideReset.value, CadDirective.MARK.some)
+      assertEquals(r1.mountGuide.mode.value.flatMap(Enumerated[BinaryOnOff].fromTag),
+                   BinaryOnOff.On.some
+      )
+      assertEquals(r1.mountGuide.source.value, "SCS".some)
+
+      assertEquals(r2.m1Guide.value.flatMap(Enumerated[BinaryOnOff].fromTag), BinaryOnOff.Off.some)
+      assertEquals(r2.m2Guide.value.flatMap(Enumerated[BinaryOnOff].fromTag), BinaryOnOff.Off.some)
+      assertEquals(r2.mountGuide.mode.value.flatMap(Enumerated[BinaryOnOff].fromTag),
+                   BinaryOnOff.Off.some
+      )
+      assertEquals(r1.guiderGains.p1TipGain.value, "0.03".some)
+      assertEquals(r1.guiderGains.p1TiltGain.value, "0.03".some)
+      assertEquals(r1.guiderGains.p1FocusGain.value, "2.0E-5".some)
```

Comment:
This is encoded with the default double encoder, will the TCS work fine with numbers encoded this way?

Reply:
Yes